server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -623,4 +623,8 @@ public PerRecipientHeaders getPerRecipientSpecificHeaders() {
     public void addSpecificHeaderForRecipient(Header header, MailAddress recipient) {
         perRecipientSpecificHeaders.addHeaderForRecipient(header, recipient);
     }
+
+    public void addAllSpecificHeaderForRecipient(PerRecipientHeaders perRecipientHeaders) {
```

Comment:
s/Header/Headers/

Reply:
Plural is hold by all, no?

add all (SpecificHeaderForRecipient)